Tests/Serialization/ResourceGraphSerializerTests.cs

```diff
@@ -0,0 +1,214 @@
+            PaginationContext paginationContext,
+            IncludingContext includingContext)
+        {
+            _output.WriteLine("Serializer type: ResourceGraphSerializer");
+            return new ResourceGraphSerializer(value, type, baseUrl, urlBuilder, paginationContext, includingContext);
+        }
+
+        public override void IncludedResourceOnlyOnce()
+        {
+            var job = new CompanyWithCustomers(id: "457", prefill: true);
+            var person = new Person(true)
+            {
+                Friends = new List<Person>
+                {
+                    new Person(id: "124", prefill: true) {
+                        Job = job
+                    },
+                    new Person(id: "125", prefill: true) {
+                        Job = job
+                    }
+                }
+            };
+
+            var include = new IncludingContext(GetQuery("include", "friends.job"));
+            var target = GetSerializer(person, DefaultResource,
+                GetUri(id: "123"), DefaultPathBuilder, null, include);
+            var result = target.Serialize();
+            _output.WriteLine(result.ToString());
+
+            var included = result["included"] as JArray;
+
+            Assert.Equal(3, included.Count);
+            /**
+             *  Changed from Assert.Equal(4, included.Count) in the original version
+             *  - an includes query of "friends.job" should only include the job of 
+             *    the person's friends, not the person's job
+             */
+        }
+
+        public override void SerializesRelationshipData()
+        {
+            var person = new PersonWithNoJob();
+            var target = GetSerializer(person, new PersonWithDefaultIdResource(),
+                GetUri(id: "123"), DefaultPathBuilder, null, null);
+            var result = target.Serialize();
+            _output.WriteLine(result.ToString());
+
+            var relationships = result["data"]["relationships"];
+            var job = relationships["job"];
+            var friends = relationships["friends"];
+
+            Assert.Equal(job["data"].Type.ToString(), "Null");
+            /**
+             *  Changed from Assert.Null(job["data"]) in the original version
+             *  - relationships that are known to be null or empty should return data representing that
+             *    an absence of the data property indicates that a client should request the link to 
+             *    expand the relationship
+             */
+
+            Assert.NotNull(friends);
+        }
+
+        public override void HandlesNullValues()
+        {
+            var person = new Person(id: "45");
+            var target = GetSerializer(person, DefaultResource,
+                GetUri(id: "45"), DefaultPathBuilder, null, new IncludingContext { DisableDefaultIncluded = true });
+            /**
+             *  Changed from a null IncludingContext in the original version
```

Comment:
No, I think the purpose of this test was to prove that if DisableDefaultIncluded was true then no relationships were returned.

It's entirely possible that I've misread this of course 😆